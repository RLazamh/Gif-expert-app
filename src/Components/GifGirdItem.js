import React from 'react'
// import PropTypes from 'prop-types'

const GifGirdItem = ({id,title,url}) => {
    // console.log(id,title,url)
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src = {url}
            alt ={title} 
            />
            <p>{title}</p>
        </div>
    )
}

// GifGirdItem.propTypes = {

// }

export default GifGirdItem

