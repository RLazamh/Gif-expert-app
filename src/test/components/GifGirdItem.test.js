import React from 'react'
import {shallow} from 'enzyme'
import GifGirdItem from "../../Components/GifGirdItem"
const title = 'Hola';
const url = 'https://s1.1zoom.me/big3/471/Painting_Art_Back_view_Photographer_575380_3840x2400.jpg'
const wrapper = shallow(<GifGirdItem title = { title } url = { url } />)

describe('Pruebas en <GifGirdItem />', ()=>{
    test('should see correctly the component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe Tener un parrafo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    })

    test('should have img al url y alt that the prototypes',()=>{
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    test('should have animate__fadeIn ', () => {
        const div = wrapper.find('div')
        expect(div.prop('className')).toMatch('animate__fadeIn')
        // forma del inge 
        const className = div.prop('className');
        // console.log(className)
        expect(`${className}`.includes('animate__fadeIn')).toBe(true)
        
    })
    
    
})