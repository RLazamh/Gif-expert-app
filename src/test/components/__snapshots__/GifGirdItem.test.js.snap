// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGirdItem /> should see correctly the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola"
    src="https://s1.1zoom.me/big3/471/Painting_Art_Back_view_Photographer_575380_3840x2400.jpg"
  />
  <p>
    Hola
  </p>
</div>
`;
